src/components/ui/Animation.js

```diff
@@ -223,7 +279,42 @@ export default function animateChart(stx, animationParameters, easeMachine) {
 
             let beginningOffset = 0;
             if (chartJustAdvanced) {
-                if (!this.animations.zoom.hasCompleted) {
+                if (this.animations.zoom.hasCompleted) {
+                    const candleWidth = this.layout.candleWidth;
+                    if (chart.scroll <= chart.maxTicks && !chart.lockScroll) {
+                        while (this.micropixels > 0) { // If micropixels is larger than a candle then scroll back further
+                            chart.scroll++;
+                            this.micropixels -= candleWidth;
+                        }
```

Comment:
```
if (this.micropixels > 0) {
    const count = Math.ceil(this.micropixels / candleWidth);
    this.scroll += count;
    this.micropixels -= count * candleWidth;
}
```

Let's remove the `while` since it could be expensive.